accessibility/README.md

```diff
@@ -0,0 +1,495 @@
+
+  <input id="search" type="text" name="search">
+  ```
+
+### Ensure interactive elements have accessible text
+
+#### Don't
+
+```vue
+<a class="site-logo" href="/">
+  <brand-logo />
+</a>
+
+<button
+  type="button"
+  @click="openMenuDrawer"
+>
+  <icon-menu />
+</button>
+
+<a href="/account">
+  <icon-account />
+</a>
+
+<a href="/sale">
+  <!-- Image containing embedded text about a 10% off sale -->
+  <img src="10-off-promo-image.jpg">
+</a>
+
+
+```
+
+#### Do
+
+```vue
+<a class="site-logo" href="/">
+  <brand-logo />
+  
+  <span class="visually-hidden">
+    My Shop Name
+  </span>
+</a>
+
+<button
+  type="button"
+  @click="openMenuDrawer"
+>
+  <icon-menu />
+  
+  <span class="visually-hidden">
+    Toggle Menu
+  </span>
+</button>
+
+<a href="/account">
+  <icon-account />
+  
+  <span class="visually-hidden">
+    Account
+  </span>
+</a>
+
+<a href="/sale">
+  <!-- Image containing embedded text about a 10% off sale -->
+  <img src="10-off-promo-image.jpg" alt="Get 10% off">
+</a>
+```
+
+* When using `<button>` or `<a>` elements containing SVG elements, ensure that accessible text is also included
+  * If the text should not be visible, the `visually-hidden` class can be used
+* If a `<button>` or `<a>` contains a `<img>` element - ensure that there is either appropriate `alt` text on the image, or add a supporting `visually-hidden` span with appropriate text.
+  * Avoid using both `alt` and `visually-hidden` text for the same element, otherwise screen readers will read out possibly duplicate or irrelevant content.
+  * Setting an empty `alt` text is fine when using `visually-hidden` text - see [Alt Text](#alt-text) for more details.
+
+### Ensure correct usage of `aria-labelledby` and `aria-label` 
+
+* The `aria-labelledby` and `aria-label` attributes provide another method for adding labels to elements
+* These are alternatives to using a `visually-hidden` element showcased in the examples above
+
+#### `aria-labelledby`
+
+* `aria-labelledby` can be used to label an element based on the text content of another existing element by it's unique `id`
+* If a heading is present inside a region, this could be a good candidate for `aria-labelledby`
+
+```html
+<section aria-labelledby="template-title">
+  <h1 id="template-title">
+    Title for the section
+  </h1>
+
+  <!-- other content -->
+</section>
+```
+* Ensure that the element referenced by `aria-labelledby` exists on the page.
+  * If the referenced element is missing, screen readers will still read out the the other text content contained in the element, but users may be missing out on the additional context that the referenced element might be adding
+* When using `aria-labelledby` in a component, consider whether the element you are referencing existing inside the component's template code.
+  * If the element you are targeting with `aria-labelledby` is stored in another component or snippet, you may not be able to guarantee that it will always be on the page.
```

Comment:
```suggestion
  * If the element you are targeting with `aria-labelledby` is stored in another component or snippet, you may not be able to guarantee that it will always be on the page
```